src/disk_interface.cc

```diff
@@ -121,7 +122,7 @@ bool StatAllFilesInDir(const string& dir, map<string, TimeStamp>* stamps,
     transform(lowername.begin(), lowername.end(), lowername.begin(), ::tolower);
     stamps->insert(make_pair(lowername,
```

Comment:
We can reduce the allocations / copies involved here (C++98 / C++03 does not have move semantics, so the call to make_pair incurs a copy)

do it this way:
```cpp
std::pair<std::string, TimeStamp*> my_pair = make_pair(WideToUtf8(ffd.cFileName), TimeStampFromFileTime(ffd.ftLastWriteTime));
transform(my_pair.first.begin(), my_pair.first.end(), my_pair.first.begin(), ::tolower);
stamps->insert(my_pair);
```

That'll reduce the total number of string allocations / copies by one.

Better, of course, would be to use C++11, and `std::move()`, but that's not currently an option.

Reply:
Isn't this outside of this PR though?

Reply:
The same "optimization" applies to both versions of the function, but this PR already makes pretty significant changes to the function, especially with regard to where it makes the most sense to place the "lifetime" variable.

If you'd like to keep things separate, then yes, this can be made into a separate PR. However, doing so will cause a merge conflict with this PR.

I can do this if you want, but I'd rather it go into this PR just for sake of saving myself the effort.

Reply:
@jhasse If using C++11 (as you mentioned in another command) is allowed inside of `#ifdef _WIN32`, then it'd be better still to replace the code with 

```cpp
std::wstring lowername = WideToUtf8(ffd.cFileName);
transform(lowername.begin(), lowername.end(), lowername.begin(), ::tolower);
stamps->emplace(std::move(lowername), TimeStampFromFileTime(ffd.ftLastWriteTime)));
```

Reply:
If possible, in a separate commit.

Reply:
https://github.com/ninja-build/ninja/pull/1678